### ISSUE TITLE
500 errors created by ErrorInternalServiceError are not logged as expected

### ISSUE THREAD

**Issue by abonander** [actix/actix-web#1557]
## Expected Behavior
Errors returned from a route using `actix_web::error::ErrorInternalServerError` should be logged:

```rust
use actix_web::{get, Response};
use actix_web::error::ErrorInternalServerError;

#[get("/path")]
fn returns_error() -> actix_web::Result<Response> {
    Err(ErrorInternalServerError("this route should print an error to the console").into())
}
```

## Current Behavior
Nothing happens. If the `Logger` middleware is configured, it reports a 500 on the route as expected but that's it; it doesn't report the text or backtrace of the error.

This is probably because logging the cause for `500 Internal Server Error` is done in `Response::from_error()` here: https://github.com/actix/actix-web/blob/2.0/actix-http/src/response.rs#L57-L59

But `InternalError` which is returned by `ErrorInternalServerError()` constructs a `Response` directly here: https://github.com/actix/actix-web/blob/2.0/actix-http/src/error.rs#L555-L564

## Possible Solution
`InternalError` should collect a backtrace when constructed and perhaps log it in its `ResponseError` impl when `error_response()` is called.

Alternatively, `Response::from_error()` should be used instead of calling `ResponseError::response_error()` directly, wherever an `actix_web::error::Error` is turned into a `Response` (I'm not sure where this is done).

## Context
We use `ErrorInternalServerError` to map `anyhow::Error` to `actix_web::Error` but with the `Logger` middleware ignoring 500s it's hard to get context on those errors right now. The error message is returned in the response which _helps_ but there's no way to get the error backtrace because it's suppressed.

## Your Environment

* Rust Version (I.e, output of `rustc -V`): rustc 1.43.0-nightly (c20d7eecb 2020-03-11)
* Actix Web Version: 2.0.0

**Comment by robjtede** [comment]
is this still a problem in v3?

**Comment by kujeger** [comment]
Behaviour seems to be the same in v4. I worked around it by using a wrapper with calls `error!(..)`, but it would be nice if the logging middleware was automatically called to log an error.

**Comment by robjtede** [comment]
more recent discussion: https://github.com/actix/actix-web/issues/2637

tl;dr: We're not going to spam error logs unconditionally with the Logger middleware. Error responses are not always unhandled errors or error-loggable events from the application perspective. The debug log is present for this reason and you can turn it on conditionally using env_logger but Logger is for Apache-style request/response logging, not error logging.

**Comment by lbfalvy** [comment]
~I get why you don't want to indiscriminately log error responses, but it would be nice to get more information than the number 500 for example if I forget to initialize some app data, especially when all the debug envvars are set both at build time and runtime.~

all the debug envvars were not infact set
